test/unit/state/action-creators.spec.js

```diff
@@ -0,0 +1,250 @@
+
+      // Phase 1: flush any existing animations
+      expect(store.dispatch).toHaveBeenCalledWith(prepare());
+      expect(store.dispatch).toHaveBeenCalledTimes(1);
+
+      // Phase 2: request dimensions after flushing animations
+      jest.runOnlyPendingTimers();
+
+      expect(store.dispatch).toHaveBeenCalledWith(requestDimensions(defaultType));
+      expect(store.dispatch).toHaveBeenCalledTimes(2);
+
+      // publishing some fake dimensions
+      store.dispatch(publishDroppableDimension(home));
+      store.dispatch(publishDraggableDimension(inHome1));
+      // now called four times
+      expect(store.dispatch).toHaveBeenCalledTimes(4);
+
+      // Phase 3: after dimensions are collected complete the lift
+      jest.runOnlyPendingTimers();
+
+      expect(store.dispatch).toHaveBeenCalledWith(completeLift(
+        liftDefaults.id,
+        liftDefaults.type,
+        liftDefaults.client,
+        liftDefaults.windowScroll,
+        liftDefaults.isScrollAllowed
+      ));
+      expect(store.dispatch).toHaveBeenCalledTimes(5);
+    });
+
+    describe('flushing previous drop animations', () => {
+      const dropAnimatingState: State = (() => {
+        const pending: PendingDrop = {
+          trigger: 'CANCEL',
+          newHomeOffset: origin,
+          impact: noImpact,
+          result: {
+            draggableId,
+            type: defaultType,
+            source: {
+              droppableId: 'drop-1',
+              index: 0,
+            },
+            destination: null,
+          },
+        };
+        const state: State = {
+          phase: 'DROP_ANIMATING',
+          drag: null,
+          drop: {
+            pending,
+            result: null,
+          },
+          dimension: noDimensions,
+        };
+        return state;
+      })();
+
+      it('should flush any existing drop animation', () => {
+        const dispatch: Function = jest.fn();
+        const getState: Function = jest.fn(() => dropAnimatingState);
+
+        liftWithDefaults()(dispatch, getState);
+
+        // $ExpectError - not checking for null in state shape
+        expect(dispatch).toHaveBeenCalledWith(completeDrop(dropAnimatingState.drop.pending.result));
+        expect(dispatch).toHaveBeenCalledWith(prepare());
+        expect(console.error).not.toHaveBeenCalled();
+      });
+
+      it('should clean the state and log an error if there is an invalid drop animating state', () => {
+        const state: State = {
+          ...initialState,
+          // hacking the phase
+          phase: 'DROP_ANIMATING',
+        };
+        const dispatch: Function = jest.fn();
+        const getState: Function = jest.fn(() => state);
+
+        liftWithDefaults()(dispatch, getState);
+
+        expect(dispatch).toHaveBeenCalledWith(clean());
+        expect(console.error).toHaveBeenCalled();
+      });
+
+      it('should not begin a lift if the drag is cancelled while the animations are flushing', () => {
```

Comment:
This test executes a current error case - which is what caused me to make this change